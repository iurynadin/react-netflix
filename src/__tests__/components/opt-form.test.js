import React from 'react'
import { render } from '@testing-library/react'
import { OptForm } from '../../components'

describe('<OptForm />', () => {
    it('redenrs the <OptForm /> with populated data', () => {
        const { container, getByText, getByPlaceholderText } = render(
            <OptForm>
                <OptForm.Input placeholder="Email address" />
                <OptForm.Button>Try it Now</OptForm.Button>
                <OptForm.Break />
                <OptForm.Text>Ready to watch?</OptForm.Text>
            </OptForm>
        );

        expect(getByText('Try it Now')).toBeTruthy();
        expect(getByText('Ready to watch?')).toBeTruthy();
        expect(getByPlaceholderText('Email address')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});