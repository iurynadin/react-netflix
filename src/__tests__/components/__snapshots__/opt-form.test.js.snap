// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> redenrs the <OptForm /> with populated data 1`] = `
<div
  class="opt-form__Container-sc-8oudnm-0 eulPEt"
>
  <input
    class="opt-form__Input-sc-8oudnm-1 diRyqI"
    placeholder="Email address"
  />
  <button
    class="opt-form__Button-sc-8oudnm-3 cJarkG"
  >
    Try it Now
     
    <img
      alt=""
      src="/images/icons/chevron-right.png"
    />
  </button>
  <div
    class="opt-form__Break-sc-8oudnm-2 hLzMWV"
  />
  <p
    class="opt-form__Text-sc-8oudnm-4 kIDSNM"
  >
    Ready to watch?
  </p>
</div>
`;
